From a3b0cb755ff08721a3b9e2c25d4aaf655fb40804 Mon Sep 17 00:00:00 2001
From: Rolf Eike Beer <kde@opensource.sf-tec.de>
Date: Wed, 6 Jun 2012 17:40:32 +0200
Subject: [PATCH] fix NULL-deref when new setting default key

Thanks to John Tapsell for spotting the line I have been looking at for weeks without seeing the bug.

CCBUG:298465
CCMAIL:johnflux@gmail.com
---
 model/kgpgitemmodel.cpp |    8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/model/kgpgitemmodel.cpp b/model/kgpgitemmodel.cpp
index 92207c6..d369346 100644
--- a/model/kgpgitemmodel.cpp
+++ b/model/kgpgitemmodel.cpp
@@ -343,8 +343,12 @@ KGpgItemModel::setDefaultKey(KGpgKeyNode *def)
 		emit dataChanged(createIndex(odefrow, 0, nd), createIndex(odefrow, lastcol, nd));
 	}
 
-	m_default = def->getId();
-	emit dataChanged(createIndex(defrow, 0, def), createIndex(defrow, lastcol, def));
+	if (def) {
+		m_default = def->getId();
+		emit dataChanged(createIndex(defrow, 0, def), createIndex(defrow, lastcol, def));
+	} else {
+		m_default.clear();
+	}
 }
 
 QModelIndex
-- 
1.7.10

